fdbclient/ReadYourWrites.actor.cpp

```diff
@@ -1134,7 +1134,7 @@ ReadYourWritesTransaction::ReadYourWritesTransaction(Database const& cx)
 
 ACTOR Future<Void> timebomb(double endTime, Promise<Void> resetPromise) {
 	while(now() < endTime) {
-		wait( delayUntil( std::min(endTime, now() + CLIENT_KNOBS->TRANSACTION_TIMEOUT_DELAY_INTERVAL) ) );
```

Comment:
Out of curiosity, how exactly can this get stuck in a loop? I feel like `delayUntil(endTime)` completing should imply `!(now() < endTime)`

Reply:
There are various floating point operations that I gather could cause a delayUntil(20.0) to fire at something like 19.999999s. I agree that a better contract from `delay` would be nice, though.

Reply:
This is also mainly a problem is simulation where delay(0) can actually happen instantly, so now()-endTime can round to 0, but now()<endTime